blocks/category-carousel-block/features/category-carousel/default.jsx

```diff
@@ -0,0 +1,108 @@
+import React from "react";
+import { useComponentContext, useFusionContext } from "fusion:context";
+import getProperties from "fusion:properties";
+import getTranslatedPhrases from "fusion:intl";
+
+import PropTypes from "@arc-fusion/prop-types";
+import { Carousel, Heading, HeadingSection, Icon, Image } from "@wpmedia/arc-themes-components";
+
+const BLOCK_CLASS_NAME = "b-category-carousel";
+const MIN_SLIDES = 4;
+const MAX_SLIDES = 12;
+
+const customFieldGroups = Array(MAX_SLIDES)
+	.fill(MIN_SLIDES)
+	.map((minimum, index) => ({
+		[`imageUrl_${index}`]: PropTypes.string.tag({
+			label: "Image",
+			description: "Select an image for this item.",
+			searchable: "image",
+			group: `Category ${index + 1}${index < minimum ? " (required)" : ""}`,
+			required: index < minimum,
+		}),
+		[`label_${index}`]: PropTypes.string.tag({
+			label: "Label",
+			description: "Enter a category label for this item.",
+			group: `Category ${index + 1}${index < minimum ? " (required)" : ""}`,
+			required: index < minimum,
+		}),
+		[`linkUrl_${index}`]: PropTypes.string.tag({
+			label: "Link Url",
+			description: "Enter a url to navigate to when selected.",
+			group: `Category ${index + 1}${index < minimum ? " (required)" : ""}`,
+			required: index < minimum,
+		}),
+	}));
+
+const mergeObjectArray = (accumulator, item) => ({ ...accumulator, ...item });
+
+function CategoryCarousel({ customFields }) {
+	const { headerText } = customFields;
+
+	const { id } = useComponentContext();
+	const { arcSite } = useFusionContext();
+	const { locale = "en" } = getProperties(arcSite);
+	const phrases = getTranslatedPhrases(locale);
+
+	const validCategoryData = customFieldGroups
+		.map((fieldDefinition) =>
+			Object.keys(fieldDefinition)
+				.map((field) => ({ [field.replace(/_\d+$/, "")]: customFields[field] }))
+				.reduce(mergeObjectArray)
+		)
+		.filter((fields) => fields.imageUrl && fields.label && fields.linkUrl);
+
+	return validCategoryData.length >= MIN_SLIDES ? (
+		<HeadingSection>
+			<div className={BLOCK_CLASS_NAME}>
+				{headerText ? <Heading>{headerText}</Heading> : null}
+				<Carousel
+					id={id}
+					label={phrases.t("category-carousel.aria-label")}
+					nextButton={
+						<Carousel.Button id={id}>
+							<Icon name="ArrowRight" />
+							{phrases.t("category-carousel.right-arrow-label")}
+						</Carousel.Button>
+					}
+					previousButton={
+						<Carousel.Button id={id}>
+							<Icon name="ArrowLeft" />
+							{phrases.t("category-carousel.left-arrow-label")}
```

Comment:
Same as other button, label should be applied to the button

Reply:
Fixed in latest